Add markup dropdown box (non-functional), polish Inventory UI

GongBingWong opened the pull request.

GongBingWong committed:
Merge pull request #53 from daramtulla/main

Update GongBingDev branch with latest main branch changes

GongBingWong committed:
Add markup dropdown box (non-functional)

In the hierarchy I created a Canvas gameobject and dragged the "InventoryItemPrefab" prefab gameobject as a child gameobject under it.

The "Markup" gameobject under "InventoryItemPrefab" is where the dropdown box is.

GongBingWong committed:
Align Inventory headings and enlarge dropdown box

GongBingWong committed:
Realign headings, Reduce Inventory name font size

I realigned name, group, and tier. I reduced inventory name font size from 6 to 5.5.

GongBingWong committed:
Fix Inventory UI text squash and bigger dropdown

To fix the Inventory UI squash I changed the y from 2 to 2.625 of all the child gameobjects (except the Markup gameobject and the inactive Static_Markup gameobject) of InventoryItemPrefab. My guess of the cause of the squished text is caused by the scaling of the parent gameobjects. For example the Content gameobject under the Viewport which is under the Inventory Scroll View which is then under the Inventory Panel has a 10.5, 4, 1 scaling (x, y, z). The Inventory Scroll View and Inventory Panel gameobjects have their own scaling too.

What I changed on InventoryItemPrefab to get bigger dropdown box: 50 markup height, 56 height content, 40 height item. I also doubled the font size in Label and Item Label.

GongBingWong committed:
Fix Glossary UI squash and shorten heights

In the Rect Transform component, the heights of both the InventoryItemPrefab and the GlossaryTermPrefab gameobjects are shortened to deal with the y scale increase.

GongBingWong committed:
Remove glTF and OBJ from Medieval Village MegaKit

GongBingWong committed:
Delete unused assets from Medieval Interior Pack

GongBingWong committed:
Fix Suppliers UI text squash and item name size

GongBingWong committed:
Add missing front and side exterior doors

GongBingWong committed:
Add new glossary title and glos, inv, sup headings

glos = glossary
inv = inventory
sup = suppliers

GongBingWong committed:
Add hiringTitle and upgradesTitle images

GongBingWong committed:
Add tax (End day screen stats broken)

I stashed away my changes on the GameManager script (so before I started working on the tax stuff) and it the end day screen was still broken.

zachj112 reviewed (commented):
Upgrade menu scaling is off. Besides that everything else looked good.

GongBingWong committed:
Add tax text and add refs to database (again)

For some reason, some of my work in the Unity scene (even though I committed) keeps getting rolled back. (Same thing happened at least once before.)

In this instance it was the references in the GameManager script missing tax text, daily sales number, and money count. Also I had to readd the TaxTitle and TaxAmount gameobjects containing the text component.

When I switched to main branch and switched back to GongBingDev, there was even more of a rollback with the "Daily Sales: Number" text missing as well. (I don't know if the "Add 100 Daily Sales" button was missing too but it might have been as well). Closing Unity and reopening it again brought that back but the stuff in the prior paragraph was still missing so hence this commit.

GongBingWong committed:
Change store and edit money value in title screen

I deleted a lot of assets in one of my commits and these assets were used in the title scene (which caused errors) so I copied the store from the store scene to use in the title screen. I couldn't copy the interior since since the pedestals were connected to scripts.

I also changed the MoneyValue gameobject's Text component's Horizontal Overflow from Wrap to Overflow to prevent the money being split into two rows (with a "0" being shown on the 2nd line with 1000.00 money).

GongBingWong committed:
Fix the "Money Made" being 0 in EndDayScreen

The "money made being 0" issue was because of "dailySales = 0;" being called in IncrDay() function which is before the EndDay() function that displays the money made. So I moved "dailySales = 0" to the GameManager script after the EndDay() function.

After that the money made never reset (kept going up) so I made changes to the timer active, but then realized it was meant to be a one time trigger, so I undid the changes. And then it randomly fixed itself? (Money reset again.) I have 0 idea what happened.

GongBingWong committed:
Get tax as red and negative in EndDayScreen

GongBingWong committed:
Small comment correction on tax at 500 money

GongBingWong committed:
Merge branch 'main' into GongBingDev

GongBingWong committed:
Reedit Inventory scaling for Content

zachj112 reviewed (approved):
Reviewed in zoom
